ksonnet-gen/ksonnet/emit.go

```diff
@@ -422,6 +429,20 @@ func (ao *apiObject) emitAsRefMixins(
 	m.writeLine("},")
 }
 
+func (ao *apiObject) emitConstructor(m *indentWriter) {
+	if dm, ok := ao.propertyMethods["default"]; ok {
```

Comment:
I'm still not a huge fan of `default` as the constructor name over something like `new` or `init`.  But we can talk about that later.